smartsim/_core/config/config.py

```diff
@@ -180,6 +180,10 @@ def test_num_gpus(self) -> int:  # pragma: no cover
     def test_port(self) -> int:  # pragma: no cover
         return int(os.environ.get("SMARTSIM_TEST_PORT", 6780))
 
+    @property
+    def test_num_ports(self) -> int:  # pragma: no cover
```

Comment:
1. I'm really disliking the mixing of test and production configuration in `config.py`. It might be better to set test values in `conftest.py` - or in this case, right in the fixture. 

2. Why do we limit ourselves here to 10 available ports? It could be 100 or 1000 if you're trying to avoid an infinite loop. If we actually need 10 databases and there is a port or two unavailable, the tests just start failing.

Reply:
1. Totally agree, we are already pushing the number of public methods on the `Config` class! I'll write up a ticket to make multiple configuration class for different areas of configuration!!
1. After talking with Andrew, it seems like you are right! The 10 port limit was deliberately chosen to prevent an infinite loop, and 10 was chosen slightly arbitrarily just to keep the iteration to an order of 10s. Sounds like there is some wiggle room to be debated, but I actually kinda like that it doesn't check an infinite number of ports as if orcs start leaking from tests we have a way to back track to the offending test. Willing to be debated on this point though!!

Reply:
I'm ok w/any number that feels less likely to randomly break than 10 - anything between 25 and 1000 is still arbitrary but also feels safer!

Reply:
That makes sense, and I like the reasoning for a higher number! I'll bump it to 25, and we can increase it in future as we see fit!